.github/workflows/amazon_linux_2023_packaging.yml

```diff
@@ -0,0 +1,172 @@
+name: Amazon Linux 2023 RPM Build and Test
+
+on:
+  pull_request_target:
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TERM_COLOR: always
+  CARGO_INCREMENTAL: 0
+  S3_BUCKET_NAME: ${{ vars.S3_BUCKET_NAME }}
+  S3_REGION: ${{ vars.S3_REGION }}
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  al2023-package-test:
+    name: Amazon Linux 2023 RPM Build and Test
+    runs-on: ubuntu-latest
+    environment: PR integration tests
+    container:
+      image: amazonlinux:2023
+      options: --privileged
+
+    steps:
+    - name: Install Git
+      run: dnf -y install git
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v5
+      with:
+        role-to-assume: ${{ vars.ACTIONS_IAM_ROLE }}
+        aws-region: ${{ vars.S3_REGION }}
+
+    - name: Checkout code
+      uses: actions/checkout@v5
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        submodules: true
+        persist-credentials: false
+
+    - name: Install build tools and dependencies
+      run: |
+        dnf -y install rpm-build rpmdevtools make git \
+                      mock \
+                      ca-certificates \
+                      python3 \
+                      rust \
+                      cargo \
+                      
+        cargo install cargo-about
+
+    - name: Generate Amazon Linux 2023 spec file
+      run: |
+        cd $GITHUB_WORKSPACE
+        cd package
+        python3 generate_spec.py amzn2023
+        ls -la amzn2023-packaging.spec
+
+    - name: Build source tarball and SRPM
+      run: |
+        cd $GITHUB_WORKSPACE
+        
+        echo " Extracting version from spec file..."
+        VERSION=$(awk '/^Version:/ {print $2}' package/amzn2023-packaging.spec)
+        echo " Building version: $VERSION"
+        
+        echo " Generating vendor dependencies..."
+        cargo vendor
+        echo " Cargo vendor completed"
+        
+        echo " Generating license file..."
+        cargo about generate --config package/attribution.toml --output-file THIRD_PARTY_LICENSES package/attribution.hbs
+        echo "Third party License file generated"
+
+        echo " Setting up RPM build directory..."
+        rpmdev-setuptree
+        echo " RPM directories created"
+
+        cp package/amzn2023-packaging.spec ~/rpmbuild/SPECS/
+
+        cp LICENSE ~/rpmbuild/SOURCES/
+        cp NOTICE ~/rpmbuild/SOURCES/
+        cp THIRD_PARTY_LICENSES ~/rpmbuild/SOURCES/
+        echo "RPM Sources created"
+
+        echo " Creating source tarball..."
```

Comment:
This dance here is confusing - it's not obvious where our cwd is at any given point